### PR TITLE
OSSM-3727 Refactor TestAuthPolicy (T18)

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/tests/tasks/security/authentication/auth_test.go [hunk 1]
```diff
@@ -16,319 +16,381 @@ package authentication
...
-			util.Inspect(err, "Failed to get sleep pod name", "", t)
-			cmd := fmt.Sprintf(`curl http://httpbin.%s:8000/ip -s -o /dev/null -w "sleep.%s to httpbin.%s: %%{http_code}"`,
-				to, from, to)
-			msg, _ := util.PodExec(from, sleepPod, "sleep", cmd, true)
-			if strings.Contains(msg, "200") {
-				t.Errorf("Global mTLS expected 000; Got response code: %s", msg)
-				log.Log.Errorf("Global mTLS expected: 000; Got response code: %s", msg)
-			} else {
-				log.Log.Infof("Response 000 as expected: %s", msg)
+	test.NewTest(t).Id("T18").Groups(test.Full, test.InterOp).Run(func(t test.TestHelper) {
```
Comment (fjglira) [inline pkg/tests/tasks/security/authentication/auth_test.go:38]: if you reference test you can directly set here "Full, InterOp"

File: pkg/tests/tasks/security/authentication/auth_test.go [hunk 2]
```diff
@@ -16,319 +16,381 @@ package authentication
...
-					}
-				} else {
-					if !strings.Contains(msg, "200") {
-						log.Log.Errorf("Namespace mTLS expected: 200; Got unexpected response code: %s", msg)
-						t.Errorf("Namespace mTLS expected: 200; Got unexpected response code: %s", msg)
-					} else {
-						log.Log.Infof("Success. Get expected response: %s", msg)
+		})
+
+		t.NewSubTest("auto mTLS").Run(func(t test.TestHelper) {
```
Comment (fjglira) [inline pkg/tests/tasks/security/authentication/auth_test.go:67]: Maybe the name of the test could be `verify auto mtls enable`

**Comment by fjglira** [comment]
LGTM, only have two minor comments

File: pkg/tests/tasks/security/authentication/auth_test.go (hunk 1 above)
Comment (luksa) [reply pkg/tests/tasks/security/authentication/auth_test.go:38]: Done. Does this make a big difference to you (it doesn't to me, but I'm fine with using a dot import for this).

**Review by fjglira — APPROVED** [review]

**Comment by fjglira** [comment]
Approved

**Pull request merged**